### PR TITLE
Reintroduce "allow more output to response following exception in forward based on wc param"

### LINKED ISSUE #23827: Exception thrown from dispatch forward can cause inconsistant error-page handling compared to other implementations
**Describe the bug**  
If a generic Exception is thrown from a dispatch forwarded JSP and is caught by the caller, Liberty limits what you can subsequently write in the response.  In Tomcat and tWAS, you can still write text of any length to the servlet response.   Liberty currently limits you to one buffer, whether hit by reaching setBufferSize() limit (default 4k) or a flush. 

The problem can be avoided by either calling setBufferSize() before the dispatch forward setting a size large enough for the additional text to be written, 
or,
changing the Exception to a ServletException.  Throwing ServletException is not subject to the single buffer limit in Liberty.

**Steps to Reproduce**  
If your servlet has this:
        protected void service(HttpServletRequest request, HttpServletResponse response)
                        throws ServletException, IOException {
                try {
                        request.getRequestDispatcher("exception.jsp").forward(request, response);                        
                    }
                catch (Throwable t) {
                        request.getRequestDispatcher("error1.jsp").forward(request, response);
                }
        }

where exception.jsp just throws Exception,
and error1.jsp attempts to write more than 4k of text, or it writes some text, does a flush, writes more text,
it is truncated in the response.

**Expected behavior**  
All text is written to the response even if it exceeds 4k or contains flushes, following a caught Exception from a dispatch-forward.  This behavior will be enabled by a Web Container custom property, so existing code using the old behavior isn't affected.

**Diagnostic information:**  
 - OpenLiberty Version: all
 - Affected feature(s):  all spec levels of servlet and jsp/pages
 - Java Version: any
 
**Additional context**

### LINKED ISSUE #24976: Re-introduce change reverted from 24004
Re-introduce change reverted from https://github.com/OpenLiberty/open-liberty/pull/24004

### PR DESCRIPTION
**Pull request by volosied**
fixes https://github.com/OpenLiberty/open-liberty/issues/23827 ( original issue) 

also fixes #24976

### CODE REVIEW

**Comment by volosied** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.el.3.0_fat,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

**Comment by posmith** [comment]
L2 message review not required

**Review by helyarp — COMMENTED** [review]
Looks good

**Comment by volosied** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.el.3.0_fat,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

File: dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/package-info.java [hunk 1]
```diff
@@ -11,5 +11,5 @@
 /**
  * @version 1.12.0
```
Comment (volosied) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/package-info.java]: Should change to 13?

**Comment by volosied** [comment]
Testing with the feature check ignores: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_piUQAPS4Ee2WYPigVf_icg

**Comment by volosied** [comment]
JSP tests passed.  Build has other defects occur and one child build failed (infrastructure).

image

**Comment by volosied** [comment]
#libby

**Review by pmd1nh — APPROVED** [review]

**Review by epj — APPROVED** [review]

**Comment by volosied** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.el.3.0_fat,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

**Comment by volosied** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.el.3.0_fat,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

**Comment by volosied** [comment]
#libby

**Comment by volosied** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.el.3.0_fat,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

**Comment by volosied** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.jsp_fat,com.ibm.ws.jsp.2.3_fat,com.ibm.ws.el.3.0_fat,com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

**Comment by volosied** [comment]
#libby

**Pull request merged**